ngraph/core/src/op/shape_of.cpp

```diff
@@ -152,6 +153,73 @@ namespace shape_of
         }
         return false;
     }
+
+    bool evaluate_bound_shape(const Node* shape_of_node,
+                              const HostTensorVector& output_values,
+                              bool is_upper)
+    {
+        const auto& input_partial_shape = shape_of_node->get_input_partial_shape(0);
+        if (input_partial_shape.rank().is_dynamic())
+            return false;
+        const auto rank = input_partial_shape.rank().get_length();
+        auto pshape_low = PartialShape::dynamic(rank), pshape_up = PartialShape::dynamic(rank);
+        for (Dimension::value_type i = 0; i < rank; ++i)
+        {
+            Interval interval = input_partial_shape[i].get_interval();
+            pshape_low[i] = interval.get_min_val();
+            pshape_up[i] = Dimension(interval.get_max_val()).is_dynamic()
+                               ? Dimension(interval.get_max_val() - 1)
```

Comment:
Because it automatically gets treated like a dynamic dimension and it raises an exception in the shape_of evaluate as dynamic dimension could not be evaluated.